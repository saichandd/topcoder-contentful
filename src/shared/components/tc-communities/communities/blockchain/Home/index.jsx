import React from 'react';
import Viewport from 'components/Contentful/Viewport';

export default function Home() {
  return (
    <Viewport
      query={{ 'fields.name': 'Blockchain Community - Home' }}
    />
  );
}
