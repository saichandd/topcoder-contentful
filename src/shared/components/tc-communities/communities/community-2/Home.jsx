/**
 * Static implementation of Home page for a community
 */

import Viewport from 'components/Contentful/Viewport';
import React from 'react';

export default function Home() {
  return (
    <Viewport
      query={{
        'fields.name': 'Community 2 - Home Page',
      }}
    />
  );
}
