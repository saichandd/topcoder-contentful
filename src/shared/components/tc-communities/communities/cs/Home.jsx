import React from 'react';
import Viewport from 'components/Contentful/Viewport';

export default function Home() {
  return (
    <Viewport
      query={{
        'fields.name': 'CS Community - Home',
      }}
    />
  );
}
