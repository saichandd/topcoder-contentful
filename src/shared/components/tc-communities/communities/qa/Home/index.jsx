/**
 * Static implementation of Home page for Wipro 2 community
 *
 * It hardcodes data which is passed to dummy components,
 * thus we disable max-len eslint rule for this file
 */
/* eslint-disable max-len */

import React from 'react';
import Section from 'components/tc-communities/Section';
import Banner from 'components/tc-communities/Banner';
import ImageText from 'components/tc-communities/ImageText';
import ResourceCard from 'components/tc-communities/ResourceCard';
import ResourceCard2 from 'components/tc-communities/ResourceCard2';
import NewsSection from 'components/tc-communities/NewsSection';
import PT from 'prop-types';

import CommunityStats from 'containers/tc-communities/CommunityStats';
import JoinCommunity from 'containers/tc-communities/JoinCommunity';

import IconRocket from '../../../../../../assets/images/tc-communities/rocket.svg';
import IconNetwork from '../../../../../../assets/images/tc-communities/network.svg';
import IconMedal from '../../../../../../assets/images/tc-communities/medal.svg';

import resourceCardStyle from '../themes/resource-card.scss';
import style from './style.scss';
import joinButtonStyle from '../themes/join-button.scss';
import textSectionStyle from '../themes/text-section.scss';

export default function Home({
  member,
  news,
}) {
  return (
    <main>
      <Banner
        title="Topcoder QA Community"
        imageSrc="/community-app-assets/themes/qa/home/header.jpg"
      >
        <div className={style.bannerText}>
          {member ? (
            'This is the place where Topcoder members can come to focus on work that involves QA.'
          ) : (
            <span>
              Sign up. &laquo;Test&raquo; your skills. Get Rewarded.
              &zwnj;
              <span className={style.highlighted}>
It&apos;s that simple!
              </span>
            </span>
          )}
        </div>
        { !member && (
          <JoinCommunity
            theme={{ link: joinButtonStyle }}
            label="Join Now"
          />
        )}
      </Banner>

      <Section theme={textSectionStyle}>
        <p>
          One of the key drivers of Digital Transformation within large
          organizations is the mass consumerization of technology. In the brave
          new ‘connected’ world, organizations are forced to adopt new
          technologies to keep pace with the digitalization.
        </p>
        <p>
          Testing is also part of this transformation and plays a critical role
          in the organization’s business strategy. It bears striking similarity
          to the ‘pit stop’ paradigm in Formula One racing. The modern pit stop
          is a fascinating choreography of technology execution and efficiency
          that separates the winners from the crowd. This is very similar to
          how we see Quality Engineering in the new world. The nirvana state of
          driving ‘Zero Cost of Quality’ which was the primary goal of
          Enterprise IT of yesteryear, has progressively transformed to drive
          ‘Quality at zero impedance’.
        </p>
        <p>
          We are on the look-out for individuals who can run our ‘pit stops’ as
          we move to our next orbits of growth. So what are you waiting for?
          Come join us, the world’s best QA community!
        </p>
      </Section>

      <Section
        theme={{
          container: style.resourcesContainer,
        }}
      >
        <ResourceCard
          icon={IconNetwork}
          title="Learn about our platform"
          text="There are many aspects to our platform...something for everyone."
          link={{
            title: 'Browse resources',
            url: 'learn',
          }}
        />
        <ResourceCard
          icon={IconMedal}
          title="What's in it for me?"
          text="Topcoder rewards participants with cash. The more deliverables you produce, the more you can make."
          link={{
            title: 'Learn about rewards',
            url: 'learn',
          }}
        />
        <ResourceCard
          icon={IconRocket}
          title="Participate in many ways"
          text="We're always running challenges and tasks.  Check back every day to see what's new."
          link={{
            title: 'Browse challenges',
            url: 'challenges',
          }}
        />
      </Section>

      <CommunityStats />

      <Section
        theme={{
          container: style.linksContainer,
        }}
      >
        <div>
          <ImageText
            title="Ramp Up"
            text="Learn about what we're working on, our environments, procedures, etc. If you want to get involved in Topcoder projects, this is a good place to start. We run challenges and tasks constantly, so we're always looking for help."
            link={{
              title: 'Start Learning',
              url: 'learn',
            }}
            imageSrc="/community-app-assets/themes/qa/home/image-ramp-up.jpg"
          />
          <ImageText
            title="Get Involved"
            text="Whether you're a QA specialist, copilot, designer, developer, or data scientist... we want you involved. Topcoder is one of the biggest customers of our own platform, so there is never a shortage of interesting things to work on."
            link={{
              title: 'Start Contributing',
              url: 'challenges',
            }}
            imageSrc="/community-app-assets/themes/qa/home/image-get-involved.jpg"
          />
        </div>
      </Section>

      <Section
        theme={{
          container: style.resourcesContainer2,
        }}
      >
        <ResourceCard2
          theme={{
            ...resourceCardStyle,
            container: style.bgBlue,
          }}
          title="State of Quality Edition 1"
          text="An investigative report into the current state of Testing, and its evolution into Quality Engineering and Assurance."
          link={{
            openNewTab: true,
            title: 'Learn More',
            url: 'http://www.wipro.com/microsite/state-of-quality-2016/index.php',
          }}
        />
        <ResourceCard2
          theme={{
            ...resourceCardStyle,
            container: style.bgGray,
          }}
          title="Why QA Needs to be Invisible and Why Engineering Quality should be Your Ultimate Goal"
          text="&laquo;Isn't Software Testing dead?&raquo; Over the last couple of years, this though (or something similar to it) seems to be the most debated, commented and argued over."
          link={{
            openNewTab: true,
            title: 'Learn More',
            url: 'http://www.dqindia.com/why-qa-needs-to-be-invisible-and-why-engineering-quality-should-be-your-ultimate-goal/',
          }}
        />
        <ResourceCard2
          theme={{
            ...resourceCardStyle,
            container: style.bgYellow,
          }}
          title="Digital Assurance Perspectives"
          text="Digital Assurance has moved focus from functional validation of &laquo;commerce&raquo; on multiple channels in silos towards an omni-channel experience."
          link={{
            openNewTab: true,
            title: 'Learn More',
            url: 'http://www.wipro.com/blogs/digital-assurance-perspectives/',
          }}
        />
        <ResourceCard2
          theme={{
            ...resourceCardStyle,
            container: style.bgGreen,
          }}
          title="Assuring a mobile future"
          text="Traditional QA methodologies unfortunately tend to fail in the mobility landscape due to various factors due to which organizations need to adapt and transform the way QA is carried out."
          link={{
            openNewTab: true,
            title: 'Learn More',
            url: 'http://www.wipro.com/blogs/assuring-a-mobile-future-part-i/',
          }}
        />
      </Section>
      <NewsSection news={news} />
    </main>
  );
}

Home.defaultProps = {
  news: [],
};

Home.propTypes = {
  member: PT.bool.isRequired,
  news: PT.arrayOf(PT.shape()),
};
