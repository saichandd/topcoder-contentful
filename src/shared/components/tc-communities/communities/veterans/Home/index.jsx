/**
 * Static implementation of Home page for Wipro 2 community
 *
 * It hardcodes data which is passed to dummy components,
 * thus we disable max-len eslint rule for this file
 */

// import ContentfulBlock from 'components/Contentful/ContentBlock';
// import ContentfulLoader from 'containers/ContentfulLoader';
import ChallengesBlock from 'components/tc-communities/ChallengesBlock';
import ImageText from 'components/tc-communities/ImageText2';
import JoinCommunity from 'containers/tc-communities/JoinCommunity';
import LoadingIndicator from 'components/LoadingIndicator';
import Modal from 'components/Modal';
import PT from 'prop-types';
import React from 'react';
import YouTubeVideo from 'components/YouTubeVideo';
import { Button } from 'topcoder-react-ui-kit';
import { Link } from 'topcoder-react-utils';

import image01 from './images/image-01.jpg';
import image02 from './images/image-02.jpg';
import image03 from './images/image-03.jpg';
import qouteAvatar from './images/qoute-avatar.jpg';

import style from './style.scss';

export default function Home({
  activeChallenges,
  baseUrl,
  isWhatIsTopcoderVideoShown,
  loadingActiveChallenges,
  setChallengeListingFilter,
  showWhatIsTopcoderVideo,
}) {
  return (
    <main>
      {
        isWhatIsTopcoderVideoShown ? (
          <Modal
            onCancel={() => showWhatIsTopcoderVideo()}
            theme={{
              container: style.whatIsTopcoderVideo,
            }}
          >
            <YouTubeVideo
              autoplay
              src="https://www.youtube.com/embed/mD12LIqdxqk"
              title="What is Topcoder?"
            />
            <div
              onClick={() => showWhatIsTopcoderVideo()}
              onKeyPress={() => showWhatIsTopcoderVideo()}
              role="button"
              styleName="closeWhatIsTopcoderVideoCross"
              tabIndex={0}
            >
&times;
            </div>
          </Modal>
        ) : null
      }
      <div styleName="style.header">
        <h1>
Topcoder for Veterans
        </h1>
        <p>
          We help military service members and veterans transition to a career
          in technology with the world&apos;s premier crowdsourcing platform.
        </p>
        <p styleName="marginTop30">
          What is Topcoder?
          &zwnj;
          <Button
            onClick={() => showWhatIsTopcoderVideo(true)}
            theme={{
              button: style.watchVideoButton,
            }}
          >
            &#9658;&emsp;Watch video
          </Button>
        </p>
        <JoinCommunity
          label="Join Now"
          theme={{
            container: style.joinButtonInHeader,
          }}
        />
      </div>

      <h1 styleName="style.ourMission">
        We help connect talented coders with organizations across the globe in
        need of your specialized skills.
      </h1>
      {/*
      <ContentfulLoader
        entryQueries={{
          content_type: 'contentBlock',
          'fields.name': 'Demo Content For Veteran',
        }}
        render={d => d.entries.matches[0].items.map(id => <ContentfulBlock id={id} />)}
        renderPlaceholder={LoadingIndicator}
      />
      <ContentfulLoader
        entryQueries={{
          content_type: 'contentBlock',
          'fields.name': 'Demo Content For Veteran',
        }}
        render={d => d.entries.matches[0].items.map(id => <ContentfulBlock id={id} />)}
        renderPlaceholder={LoadingIndicator}
      />
      */}
      <div styleName="style.imageTextBlock">
        <ImageText imageUrl={image01}>
          <h1>
Why We Started
          </h1>
          <p>
            The U.S. military has some of the most sophisticated technology
            platforms and equipment on the planet, which demand the brightest
            minds available. And with 200,000 service members exiting the
            military annually, it’s time to apply that expertise to civilian
            employment with thousands of organizations around the world.
          </p>
          <p>
            But the transition from military to civilian life can be
            challenging and, as with any training, the most effective method
            includes real-world experience. Yet, there are few private sector
            programs outside the realm of government service that effectively
            and repeatedly provide new, realistic opportunities for service
            members. In fact, the Center for New American Security reports that
            60% of employers see the translation gap as the biggest obstacle to
            bringing more vets on board.
          </p>
          <p>
            We’re here to bridge that gap.
          </p>
        </ImageText>
        <ImageText imageUrl={image02}>
          <h1>
How the Topcoder Veterans Community Works
          </h1>
          <p>
            The challenges on veterans.topcoder.com are real-world problems and
            projects requested by Topcoder customers. Whether a contestant wins
            a challenge or not, participation demonstrates skill with current,
            in-demand toolsets. Members can compete to gain experience, and as
            proficiency increases, earn money — all from home.
          </p>
          <p>
            Topcoder also partners with advocacy groups such as OperationCode,
            a 503(c) dedicated to helping veterans learn to code. Together with
            our partners, we have developed a coalition of companies that share
            our vision and are committed to providing job opportunities to
            members of the Topcoder Veterans Community.
          </p>
          <p>
            Sign up today! Not only is it fast and free to join the Topcoder
            Veterans Community, but doing so also enables you to compete in
            interesting projects, earn money, hone your skills, and help many
            of today’s top businesses solve real-world problems.
          </p>
          <JoinCommunity
            label="Join Now"
            theme={{
              container: style.joinButtonInCard,
            }}
          />
        </ImageText>
        <ImageText imageUrl={image03}>
          <h1>
The Topcoder Team is Here to Help
          </h1>
          <p>
            If you&apos;re new to Topcoder, we&apos;re here to help. If you
            have any questions about joining or competing, contact us
            &zwnj;
            <Link
              to="mailto:support@topcoder.com"
            >
here
            </Link>
.
          </p>
          <p>
            The
            &zwnj;
            <Link
              openNewTab
              to="https://apps.topcoder.com/forums/"
            >
Topcoder forums
            </Link>
&zwnj;
            have lots of great insight and they are a good place to find
            answers to your questions.
          </p>
          <p>
            We also have Slack for our community members. Instant feedback and
            chatting opportunities with some of the world&apos;s best.
            &zwnj;
            <Link
              openNewTab
              to="https://docs.google.com/forms/d/e/1FAIpQLScb9pWbauhQ6owZZheM15f1Rodu_6CdGNtivvDb71kGgpJJKw/viewform"
            >
Topcoder Community Slack
            </Link>
.
          </p>
          <p>
            The best place to start after you register is to check the
            &zwnj;
            <Link
              to={`${baseUrl}/challenges`}
            >
challenge listings page
            </Link>
. We event have some fun challenges
            which are perfect for getting your feet wet!
          </p>
        </ImageText>
      </div>

      <div styleName="style.quoteBlock">
        <div styleName="style.qouteAvatar">
          <img alt="Qoute avatar" src={qouteAvatar} />
          <div styleName="style.qouteNote">
            <strong>
Jarah Meador
            </strong>
            {' '}
Innovation Crowdsourcing Lead at the
            VA
          </div>
        </div>
        <div>
          <div styleName="style.qouteText">
            <span styleName="style.qouteCommas style.left">
&ldquo;
            </span>
            The launch of the Topcoder Veterans Community represents a major milestone
            in our ongoing mission to help veterans develop new skills and successfully
            transition to meaningful civilian jobs.  This community is helping the VA, as
            well as other organizations, accelerate technology innovation development
            while supporting important workforce development for Veterans.
            <span styleName="style.qouteCommas style.right">
&rdquo;
            </span>
          </div>
          <div styleName="style.qouteNote">
            <strong>
Jarah Meador
            </strong>
            {' '}
Innovation Crowdsourcing Lead, U.S. Department
            of Veterans Affairs
          </div>
        </div>
      </div>
      <div styleName="style.challengesBlock">
        {
          loadingActiveChallenges && !activeChallenges ? <LoadingIndicator /> : (
            <ChallengesBlock
              baseUrl={baseUrl}
              challenges={activeChallenges}
              setChallengeListingFilter={setChallengeListingFilter}
            />
          )
        }
      </div>
      <div styleName="style.cardsBlock">
        <div styleName="style.card style.card-01">
          <div>
            <h1>
Learn & Explore
            </h1>
            <p>
              Grow your skills and get Topcoder-certified&nbsp;&mdash; all for
              free
            </p>
            <Button to={`${baseUrl}/learn`}>
Learn More
            </Button>
          </div>
        </div>
        <div styleName="style.card style.card-02">
          <div>
            <h1>
Compete & Earn
            </h1>
            <p>
              Check out our challenges and get paid for winning results
            </p>
            <Button to={`${baseUrl}/learn`}>
Learn More
            </Button>
          </div>
        </div>
      </div>
    </main>
  );
}

Home.propTypes = {
  activeChallenges: PT.arrayOf(PT.shape({

  })).isRequired,
  baseUrl: PT.string.isRequired,
  isWhatIsTopcoderVideoShown: PT.bool.isRequired,
  loadingActiveChallenges: PT.bool.isRequired,
  setChallengeListingFilter: PT.func.isRequired,
  showWhatIsTopcoderVideo: PT.func.isRequired,
};
