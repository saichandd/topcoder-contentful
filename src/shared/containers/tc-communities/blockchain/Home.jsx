import _ from 'lodash';
import Home from 'components/tc-communities/communities/blockchain/Home';
import qs from 'qs';
import { connect } from 'react-redux';

function mapStateToProps(state, props) {
  let q = props.location.search.slice(1);
  if (q) q = qs.parse(q);
  return {
    showBuffaloHackathonBanner: q.utm_campaign === 'buffalohack',
    userGroups: _.get(state.auth.profile, 'groups'),
  };
}

export default connect(mapStateToProps)(Home);
