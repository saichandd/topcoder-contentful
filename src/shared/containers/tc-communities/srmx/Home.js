import Home from 'components/tc-communities/communities/srmx/Home';
import { connect } from 'react-redux';

function mapStateToProps(state) {
  return {
    tokenV2: state.auth.tokenV2,
  };
}

export default connect(mapStateToProps)(Home);
