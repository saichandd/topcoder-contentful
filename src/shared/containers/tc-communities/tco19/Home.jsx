import _ from 'lodash';
import Home from 'components/tc-communities/communities/tco19/Home';
import { connect } from 'react-redux';

function mapStateToProps(state) {
  return {
    userGroups: _.get(state.auth.profile, 'groups'),
  };
}

export default connect(mapStateToProps)(Home);
