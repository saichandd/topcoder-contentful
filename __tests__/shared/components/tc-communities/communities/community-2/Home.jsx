import React from 'react';
import Renderer from 'react-test-renderer/shallow';
import Home from 'components/tc-communities/communities/community-2/Home';

test('Match shadow snapshot', () => {
  const renderer = new Renderer();

  renderer.render((
    <Home news={[]} />
  ));
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
