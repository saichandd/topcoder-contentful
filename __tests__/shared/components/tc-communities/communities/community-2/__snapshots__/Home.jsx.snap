// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match shadow snapshot 1`] = `
<ViewportLoader
  id={null}
  preview={false}
  query={
    Object {
      "fields.name": "Community 2 - Home Page",
    }
  }
/>
`;
